src/compiler/factory/nodeFactory.ts

```diff
@@ -7449,13 +7448,17 @@ function mergeEmitNode(sourceEmitNode: EmitNode, destEmitNode: EmitNode | undefi
     // `leadingComments` are concatenated with any existing leading comments on the destination
     if (leadingComments) {
         // We use `.slice()` in case `destEmitNode.leadingComments` is pushed to later
-        destEmitNode.leadingComments = addRange(leadingComments.slice(), destEmitNode.leadingComments);
+        for (const comment of leadingComments.slice()) {
+            destEmitNode.leadingComments = appendIfUnique(destEmitNode.leadingComments, comment);
+        }

```

Comment:
This doesn't seem like the right fix; what if someone is intentionally duplicating a comment?

I would expect the fix to take place somewhere other than where we add comments, rather something in the emitter.

Reply:
I understand but looks suspicious to me that we want to append the same comment. This is also consistent with the on the rest of the properties, i.e `helpers`.

I can check for another solution if you still think this is not correct.

Reply:
`addSyntheticLeadingComment` and friends are public APs, so this would be a breaking change to the API's behavior (we'd stop emitting comments as given, and start doing this editorializing) - we really should check this only at a higher level, while we're building the node ourselves.

Reply:
Helpers are not really comparable here; those are special functions emitted at the top of the file where you only ever want one of them at a time. They're not unique to any specific node.

But, I guess here, it's `appendIfUnique` where unique is by object identity, so it's less dangerous, but it still sort of feels like a bandaid and that there's a bug somewhere else which is causing the same comment nodes to appear in two different places.

Reply:
Sent https://github.com/microsoft/TypeScript/pull/59613 after our quick call, really wonder why the function was written this way originally?

Reply:
Correction, it looks like the issue might stem from either `deepCloneOrReuseNode` or `setTextRange` in `createNodeBuilder()`. Both `range` and `location` already have leading comments set before `setOriginalNode` is called.

Reply:
It looks like the issue is that `range` already has an `original` node pointing to `location` when we enter `setTextRange` (and thus already has the leading synthetic comments), but then we clone the range which produces a fresh copy of the node that already has its `original` node set to the previous range we cloned. 

We then end up calling `setOriginalNode` *again* to overwrite the `original` node for the `range` to point it at `location`.

The question is whether we need the call to `setOriginalNode` in `setTextRange` on L6062 of checker.ts, or if we should at least guard against calling it if we cloned it on L6060, or if it already has an `original` pointer set.

Reply:
See #59647 for the fix. We can either take that and rebase, or you can apply that change here instead.